### ISSUE TITLE
content field xss vulns

### ISSUE THREAD

**Issue by tablatronix** [GetSimpleCMS/GetSimpleCMS#1068]
content fields are susceptible to xss. But we also allow it, aside for filtering all js, which is rather difficult, what other options are there ?

**Comment by cnb** [comment]
http://htmlpurifier.org/

https://codex.wordpress.org/Function_Reference/wp_kses

**Comment by tablatronix** [comment]
Yeah i think this should be a plugin , most of these libraries are enormous. I have researched most of them.

**Comment by tablatronix** [comment]
However we do now have cleanHtml function that accepts whitelist tags, so that is fairly easy to implement. Of course its slow since it DOMs the entire string and loops it. But it can be very advanced with regards to checking and filtering.

**Comment by cnb** [comment]
Fitering (with libraries or whatever) could be done just when saving a page, by default (with option to disable).
In the frontend, filtering could be better be disabled by default (with option to enable).

Yes, this could be done by a plugin.

But, for the frontend: It may be convenient that filtering is done before running the 'content' filter. This would prevent removal of legitimate scripts, inserted by plugins (components called by DynPages or other placeholders).

**Comment by tablatronix** [comment]
Indeed, but the distribution alone of it with GS would probably increase its size significantly in core was another concern.

**Comment by tablatronix** [comment]
adding content in filter for stuff like this.
could be used for other stuff like converting short-codes or text expansions permanently, but overuse of it is probably not recommended.

**Comment by tablatronix** [comment]
@filer typo fixed
